### PR TITLE
HttpClient for ios / mac.

### PR DESCRIPTION
**Pull request by baulig**

### CODE REVIEW

File: mcs/class/System.Net.Http/NSUrlSessionHandler.cs [hunk 1]
```diff
@@ -592,7 +592,8 @@ class NSUrlSessionDataTaskStreamContent : StreamContent
 		{
 			Action disposed;
 
-			public NSUrlSessionDataTaskStreamContent (NSUrlSessionDataTaskStream source, Action onDisposed, CancellationToken token) : base (source, token)
+			[Obsolete ("FIXME: Please talk to Martin about this; see https://github.com/mono/mono/issues/12996.")]
```
Comment (marek-safar) [inline mcs/class/System.Net.Http/NSUrlSessionHandler.cs]: This is private API used in this file only in our source code, if you need to fix it then please do it

Comment (baulig) [reply mcs/class/System.Net.Http/NSUrlSessionHandler.cs]: That's why this is still a _Draft_ pull request ;-)

**Comment by lewing** [comment]
@baulig does this still represent work in progress? or should it be closed?

**Comment by baulig** [comment]
I still want to eventually finish this some day, but would then probably start on top of current master.  Closing for the moment.

**Pull request closed without merging**